### PR TITLE
models: bucket removal addition

### PR DESCRIPTION
**Pull request by lnielsen**

### CODE REVIEW

File: invenio_files_rest/models.py [hunk 1]
```diff
@@ -433,9 +433,16 @@ def delete(cls, bucket_id):
 
     @ensure_unlocked()
     def remove(self):
-        """Permanently remove a bucket and all objects."""
-        # 1) remove all object versions 2) remove bucket.
-        raise NotImplementedError()
+        """Permanently remove a bucket and all objects (including versions).
+
+        Note the method does not remove the associated file instances which
```
Comment (jirikuncar) [inline invenio_files_rest/models.py]: Shall we enhance the note with `.. note::`?

Comment (lnielsen) [inline invenio_files_rest/models.py]: Not necessary IMHO as it's doesn't need a framed box around it in the rendered documentation.

**Comment by slint** [comment]
:shipit:

File: invenio_files_rest/models.py (hunk 1 above)
Comment (jirikuncar) [inline invenio_files_rest/models.py]: There is one in `BucketTag.remove` method.

Comment (lnielsen) [inline invenio_files_rest/models.py]: ``BucketTag.remove`` -> ``ObjectVersion.remove()``?

With:

> Shall we enhance the note with .. note::?

Did you mean:

1)
```
.. note:: 

   Note the method does not remove the associated file instances which 
   must be garbage collected.
```

or:

2)
```
.. note::

   This should only be used when you want to permanently remove a bucket.
```

Comment (jirikuncar) [inline invenio_files_rest/models.py]: Kind of ...

Comment (lnielsen) [inline invenio_files_rest/models.py]: ?

Comment (jirikuncar) [inline invenio_files_rest/models.py]: Both joined as one `.. note::`? I was mostly wondering why one remove method need framed box and the other doesn't when both messages are quite important.

Comment (lnielsen) [inline invenio_files_rest/models.py]: Amended in #92

**Pull request merged**